PWGHF/Utils/utilsAnalysis.h

```diff
@@ -111,32 +109,46 @@ bool isSelectedTrackTpcQuality(T const& track, const int tpcNClustersFoundMin, c
+  o2::framework::Configurable<float> ptMassCut3ProngMax{"ptMassCut3ProngMax", 8., "Max pt to apply pt differential cut for 3 prong candidates"};
+  o2::framework::Configurable<float> ptMassCut2ProngMax{"ptMassCut2ProngMax", 8., "Max pt to apply pt differential cut for 2 prong candidates"};
   o2::framework::Configurable<o2::framework::LabeledArray<float>> deltaMassPars3Prong{"deltaMassPars3Prong", {defaultDeltaMassPars3Prong[0], 2, {"constant", "linear"}}, "delta mass parameters for HF 3 prong trigger mass cut"};
+  o2::framework::Configurable<o2::framework::LabeledArray<float>> deltaMassPars2Prong{"deltaMassPars2Prong", {defaultDeltaMassPars2Prong[0], 2, {"constant", "linear"}}, "delta mass parameters for HF 2 prong trigger mass cut"};
   o2::framework::Configurable<o2::framework::LabeledArray<float>> sigmaPars3Prong{"sigmaPars3Prong", {defaultSigmaPars3Prong[0], 2, {"constant", "linear"}}, "sigma parameters for HF 3 prong trigger mass cut"};
+  o2::framework::Configurable<o2::framework::LabeledArray<float>> sigmaPars2Prong{"sigmaPars2Prong", {defaultSigmaPars2Prong[0], 2, {"constant", "linear"}}, "sigma parameters for HF 2 prong trigger mass cut"};
 
-  /// Mass selection of D+ candidates to build B0 candidates
-  /// \param pTrackSameChargeFirst is the first same-charge track momentum
-  /// \param pTrackSameChargeFirst is the second same-charge track momentum
-  /// \param pTrackSameChargeFirst is the opposite charge track momentum
-  /// \param ptD is the pt of the D+ meson candidate
-  /// \return true if D+ passes selection
-  bool isSelectedDplusInMassRange(const float& invMassDplus, const float& ptD)
+  /// Mass selection of 2 or 3 prong canidates in triggered data analysis
+  /// \param invMass is the invariant mass of the candidate
+  /// \param pdgMass is the pdg Mass of the candidate particle
+  /// \param pt is the pt of the candidate
+  /// \return true if candidate passes selection
+  template <bool is3Prong>
+  bool isCandidateInMassRange(const float& invMass, const float& pdgMass, const float& pt)
   {
-    float peakMean = (ptD < pTMaxDeltaMass) ? ((o2::constants::physics::MassDPlus + deltaMassPars3Prong->get("constant")) + deltaMassPars3Prong->get("linear") * ptD) : o2::constants::physics::MassDPlus;
-    float peakWidth = sigmaPars3Prong->get("constant") + sigmaPars3Prong->get("linear") * ptD;
-
-    if (std::fabs(invMassDplus - peakMean) > nSigma * peakWidth && ptD < pTMaxMassCut) {
-      return false;
+    float peakMean{0.};
+    float peakWidth{0.};
+    float ptMassCutMax{0.};
+    if constexpr (is3Prong) {
+      peakMean = (pt < ptDeltaMass3ProngMax) ? ((pdgMass + deltaMassPars3Prong->get("constant")) + deltaMassPars3Prong->get("linear") * pt) : pdgMass;
+      peakWidth = sigmaPars3Prong->get("constant") + sigmaPars3Prong->get("linear") * pt;
+      ptMassCutMax = ptMassCut3ProngMax;
```

Comment:
You can reduce the branching to its minimum by assigning `nSigmaMax`, `ptDeltaMassMax`, `deltaMassPars`, `sigmaPars` in the way you do it for `ptMassCutMax`. The code itself will then work for both scenarios.